app/main.py

```diff
@@ -1,2 +1,65 @@
+from typing import Any
+
+
 class Dictionary:
-    pass
+    def __init__(self, initial_capacity: int = 8) -> None:
+        self.capacity = initial_capacity
+        self.size = 0
+        self.storage = [None] * self.capacity
+
+    class Node:
+        def __init__(self, key: Any, value: Any) -> None:
+            self.key = key
+            self.value = value
+            self.next = None
+
+    def _resize(self) -> None:
+        new_capacity = self.capacity * 2
+        new_storage = [None] * new_capacity
+
+        for i in range(self.capacity):
+            node = self.storage[i]
+            while node:
+                new_index = hash(node.key) % new_capacity
+                new_node = self.Node(node.key, node.value)
+                new_node.next = new_storage[new_index]
+                new_storage[new_index] = new_node
+                node = node.next
+
+        self.capacity = new_capacity
+        self.storage = new_storage
+
+    def __setitem__(self, key: Any, value: Any) -> None:
```

Comment:
When setting an existing key to a new value, you correctly update node.value. However, you also continue searching the rest of the chain, which is unnecessary and can lead to performance inefficiencies.